src/main/java/io/opentracing/contrib/global/GlobalSpanManager.java

```diff
@@ -0,0 +1,114 @@
+    private static final Logger LOGGER = Logger.getLogger(GlobalSpanManager.class.getName());
+
+    /**
+     * Overridable singleton instance of the global span manager.
+     */
+    private static final AtomicReference<GlobalSpanManager> INSTANCE = new AtomicReference<GlobalSpanManager>();
+
+    private static final Callable<GlobalSpanManager> DEFAULT_PROVIDER = new Callable<GlobalSpanManager>() {
+        public GlobalSpanManager call() throws Exception {
+            return new ThreadLocalSpanManager();
+        }
+    };
+
+    private static GlobalSpanManager getInstance() {
+        GlobalSpanManager instance = INSTANCE.get();
+        if (instance == null) {
+            final GlobalSpanManager singleton = ServiceLoader.loadSingleton(GlobalSpanManager.class, DEFAULT_PROVIDER);
+            while (instance == null && singleton != null) {
+                INSTANCE.compareAndSet(null, singleton);
+                instance = INSTANCE.get();
+            }
+            LOGGER.log(Level.FINE, "Singleton GlobalSpanManager implementation: {0}.", instance);
+        }
+        return instance;
+    }
+
+    /**
+     * This method allows explicit registration of a configured <code>GlobalSpanManager</code> implementation
+     * to override the behaviour of the default <code>ThreadLocal</code> implementation.
+     *
+     * @param instance The overridden implementation to use for in-process span management.
+     * @return The previous <code>GlobalSpanManager</code> that was initialized before,
+     * or <code>null</code> if there was no previous instance active.
+     */
+    protected static GlobalSpanManager registerInstance(GlobalSpanManager instance) {
+        return INSTANCE.getAndSet(instance);
+    }
+
+    /**
+     * Static method to return the currently active global {@link Span}.<br>
+     * There is no guarantee that there is an active {@link Span} in all situations.
+     *
+     * @return The currently active global Span, or <code>null</code> if there is no Span currently active.
+     */
+    public static Span activeSpan() {
+        return getInstance().getActiveSpan();
+    }
+
+    /**
+     * This method makes the specified span the <em>globally active</em> span within the global span manager.
+     * <p>
+     * It is left to the actual implementation how to deal with <code>null</code> spans.
+     * The default <code>ThreadLocal</code>-based implementation temporarily <em>suspends</em> the span activation,
+     * meaning no {@link #activeSpan()} is returned until the {@link Closeable} is closed.
+     *
+     * @param span The span to become the globally active span.
+     * @return The object that will return the <em>active</em> span back to the current state when closed.
+     * @see #activeSpan()
+     */
+    public static Closeable activate(Span span) {
+        return getInstance().setActiveSpan(span);
+    }
+
+    /**
+     * This method clears <em>all</em> active spans if they exist.
```

Comment:
maybe note why you would do this, not how implementations are likely to implement it.